File: Creational/Factory Method/ConcreteFactoryA.cpp
#include "ConcreteFactoryA.h"
#include "ConcreteProductA.h"

ConcreteFactoryA::ConcreteFactoryA()
{
}

ConcreteFactoryA::~ConcreteFactoryA()
{
}
	
AbstractProduct* ConcreteFactoryA::factory_method() const
{
	return new ConcreteProductA();
};

File: Creational/Factory Method/Golang/ConcreteFactoryA.go
package main

type ConcreteFactoryA struct{

}

func (ConcreteFactoryA) factoryMethod() AbstractProduct{
	return &ConcreteProductA{}
}

func (concreteFactoryA ConcreteFactoryA) operation(){
	var abstractProduct AbstractProduct = concreteFactoryA.factoryMethod()
	abstractProduct.operation()
}